include/ua_plugin_access_control.h

```diff
@@ -88,6 +88,31 @@ struct UA_AccessControl {
     UA_Boolean (*allowDeleteReference)(UA_Server *server, UA_AccessControl *ac,
                                        const UA_NodeId *sessionId, void *sessionContext,
                                        const UA_DeleteReferencesItem *item);
+
```

Comment:
This a pretty detailed API. And it will become a lot more verbose when more features are added.
Can we model the API according to the DataType definitions for HistoryUpdate?

Part 11 defines possible content for HistoryUpdateDetails:
- UpdateDataDetails
- UpdateStructureDataDetails
- UpdateEventDetails
- DeleteRawModifiedDetails
- DeleteAtTimeDetails
- DeleteEventDetails

Reply:
Ok, I see your problem here.

But the details corner is not the right place to connect the access control.

In UpdateDataDetails you can have more than one update.  For example 3 times a replace and 2 updates and 1 insert. With that you can not say: Insert is OK and replace is not.......

But I found a way which suites better, so we have only one access function for UpdateDataDetails and DeleteRawModifiedDetails.